File: problems/3137_minimum_number_of_operations_to_make_word_k_periodic/Solution.java
import java.util.*;

class Solution {
    public int minimumOperationsToMakeKPeriodic(String word, int k) {
        Map<String, Integer> cnt = new HashMap<>();
        int total = 0;
        for (int i = 0; i < word.length(); i += k) {
            String s = word.substring(i, i + k);
            cnt.merge(s, 1, Integer::sum);
            total++;
        }
        return total - Collections.max(cnt.values());
    }
}

File: problems/3137_minimum_number_of_operations_to_make_word_k_periodic/solution.py
class Solution:
    def minimumOperationsToMakeKPeriodic(self, word: str, k: int) -> int:
        from collections import Counter
        n = len(word)
        chunks = [word[i:i+k] for i in range(0, n, k)]
        cnt = Counter(chunks)
        return len(chunks) - max(cnt.values())
